cmd/podman/ps.go

```diff
@@ -405,94 +449,95 @@ func (p *psTemplateParams) headerMap() map[string]string {
 	return values
 }
 
+func sortPsOutput(sortBy string, psOutput psSorted) (psSorted, error) {
+	switch sortBy {
+	case "id":
+		sort.Sort(psSortedId{psOutput})
+	case "image":
+		sort.Sort(psSortedImage{psOutput})
+	case "command":
+		sort.Sort(psSortedCommand{psOutput})
+	case "runningfor":
+		sort.Sort(psSortedRunningFor{psOutput})
+	case "status":
+		sort.Sort(psSortedStatus{psOutput})
+	case "size":
+		sort.Sort(psSortedSize{psOutput})
+	case "names":
+		sort.Sort(psSortedNames{psOutput})
+	case "created":
+		sort.Sort(psSortedCreated{psOutput})
+	default:
+		return nil, errors.Errorf("invalid option for --sort, options are: command, created, id, image, names, runningfor, size, or status")
+	}
+	return psOutput, nil
+}
+
 // getTemplateOutput returns the modified container information
-func getTemplateOutput(containers []*libpod.Container, opts batchcontainer.PsOptions) (psSorted, error) {
+func getTemplateOutput(psParams []psJSONParams, opts batchcontainer.PsOptions) ([]psTemplateParams, error) {
 	var (
-		psOutput     psSorted
+		psOutput     []psTemplateParams
 		status, size string
 		ns           *batchcontainer.Namespace
 	)
+	// If the user is trying to filter based on size, or opted to sort on size
+	// the size bool must be set.
+	if strings.Contains(opts.Format, ".Size") || opts.Sort == "size" {
```

Comment:
Would strings.HasSuffix(opts.Format, ".Size") be better here?

Reply:
I disagree, because there's no guarentee the user wants the size to be the last parameter. They could want the size listed first, which HasSuffix would then return false for.